### DISCUSSION TITLE
Binding dynamic datatable to DataGrid

### DISCUSSION THREAD

**Discussion by trrahul** [AvaloniaUI/Avalonia#11785] [Q&A] [answered]
I was trying to port a WPF application to Avalonia and tried to bind a Datatable to DataGrid, but it does not display the actual table. Do you know if this feature is currently supported? If it is not, is there any other way I can bind the tabular data to the DataGrid? Maybe after converting the table to a Dictionary?

**Comment by maxkatz6** [comment]
Datatable bindings are not supported. You can bind Items/ItemsSource and populate Columns collection with columns if you need to generate them from the code.

**Comment by Dutch-0** [comment]
Would it be maybe be possible to use a custom `MyTable` class, which itself would have a `List<MyColumn>` and `List<MyRow>`?
I tried everything I could think of, including: ITypedList, CustomTypeDescriptor and IReflectableType

**Reply by timunie** [reply, marked as answer]
@Dutch-0 maybe TreeDataGrid is what you need. It can have more flexible data definitions.

**Reply by Dutch-0** [reply]
Thank you very much, just what I needed! :+1:

**Comment by sccombp** [comment]
I realise this has already been answered but here's an quick and (very) nasty example of binding a datagrid to a datatable.

[AvaloniaDatagridBindingToDatatable.zip](https://github.com/AvaloniaUI/Avalonia/files/12311475/AvaloniaDatagridBindingToDatatable.zip)

**Reply by wosiyourfather** [reply]
How to change the background color or font color based on the values in it

**Reply by sccombp** [reply]
This might do what you're after [AvaloniaDatagridBindingToDatatableColour.zip](https://github.com/AvaloniaUI/Avalonia/files/12424815/AvaloniaDatagridBindingToDatatableColour.zip)

best I could do in 5 minutes before I head off into meetings

**Comment by wosiyourfather** [comment]
> I realise this has already been answered but here's an quick and (very) nasty example of binding a datagrid to a datatable.
> 
> [AvaloniaDatagridBindingToDatatable.zip](https://github.com/AvaloniaUI/Avalonia/files/12311475/AvaloniaDatagridBindingToDatatable.zip)
How to change the background color or font color based on the values in it

**Comment by qq908382818** [comment]
DataTable dt1 = new();
  dt1.Columns.Add("row1", typeof(string));
  dt1.Columns.Add("row2", typeof(string));
  for(int i = 0; i < 10; i++)
  {
      DataRow dr1 = dt1.NewRow();
      dr1["row1"] = "row1_" + i;
      dr1["row2"] = "row2_" + i;
      dt1.Rows.Add(dr1);
  }
  this.dg1.ItemsSource = dt1.DefaultView;

<DataGrid x:Name="dg1" AutoGenerateColumns="False">
	<DataGrid.Columns>
		<DataGridTextColumn Header="row1"   Binding="{Binding Path=row1}"/>
		<DataGridTextColumn Header="row2" Binding="{Binding row2}" />
	</DataGrid.Columns>
</DataGrid>

all are wrong?how to write??
